scripts/fetch-projects.sh

```diff
@@ -0,0 +1,206 @@
+            const url = `https://api.github.com/users/${owner}/repos?type=public&per_page=${perPage}&page=${page}&sort=updated`;
+            console.log(`Fetching page ${page}...`);
+            
+            const response = await fetch(url, { headers });
+            
+            if (!response.ok) {
+                throw new Error(`Failed to fetch repos (page ${page}): ${response.status} ${response.statusText}`);
+            }
+            
+            const repos = await response.json();
+            
+            if (repos.length === 0) {
+                hasMore = false;
+            } else {
+                allRepos = allRepos.concat(repos);
+                page++;
+                
+                // GitHub API rate limiting - add small delay between requests
+                if (page > 1) {
+                    await new Promise(resolve => setTimeout(resolve, 100));
+                }
+            }
+        }
+        
+        console.log(`Found ${allRepos.length} total repositories across ${page - 1} pages`);
+        
+        const projects = [];
+        
+        for (const repo of allRepos) {
+            // Skip forks unless they have significant activity
+            if (repo.fork && repo.stargazers_count === 0) continue;
+            
+            // Skip the indietasten.github.io repository itself
+            if (repo.name === 'indietasten.github.io') continue;
+            
+            // Determine status based on repository topics
+            let status = 'in-progress'; // Default status
+            
+            const topics = repo.topics || [];
+            
+            if (topics.includes('status-done')) {
+                status = 'done';
+            } else if (repo.archived) {
+                status = 'abandoned';
+            } else if (repo.pushed_at) {
+                const lastPush = new Date(repo.pushed_at);
+                const sixMonthsAgo = new Date();
+                sixMonthsAgo.setMonth(sixMonthsAgo.getMonth() - 6);
+                
+                if (lastPush < sixMonthsAgo) {
+                    status = 'abandoned';
+                }
+            }
+            
+            // Extract tags from topics (excluding status tags)
+            const tags = topics.filter(topic => !topic.startsWith('status-'));
+            
+            // Add primary language as a tag if available
+            if (repo.language && !tags.includes(repo.language)) {
+                tags.unshift(repo.language);
+            }
+            
+            // Try to get repository's actual OpenGraph image first
+            let ogImage = null;
+            
+            // Use the dynamic OpenGraph image as default
+            ogImage = `https://opengraph.githubassets.com/1/${repo.full_name}`;
+            
+            // Note: The repository's actual social preview image would require additional API calls
+            // For now, we use the dynamic OpenGraph image which GitHub generates automatically
```

Comment:
Actually implement the API calls, or make a HEAD request to the repo url to retrieve the og image url